DQM/EcalMonitorTasks/interface/GpuTask.h

```diff
@@ -0,0 +1,104 @@
+#ifndef DQM_EcalMonitorTasks_GpuTask_H
+#define DQM_EcalMonitorTasks_GpuTask_H
+
+#include "DQWorkerTask.h"
+
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+
+namespace ecaldqm {
+
+  class GpuTask : public DQWorkerTask {
+  public:
+    GpuTask();
+    ~GpuTask() override = default;
+
+    void addDependencies(DependencySet&) override;
+
+    bool filterRunType(short const*) override;
+
+    void beginEvent(edm::Event const&, edm::EventSetup const&, bool const&, bool&) override;
+    bool analyze(void const*, Collections) override;
+
+    template <typename DigiCollection>
+    void runOnCpuDigis(DigiCollection const&, Collections);
+    template <typename DigiCollection>
+    void runOnGpuDigis(DigiCollection const&, Collections);
+    void runOnCpuUncalibRecHits(EcalUncalibratedRecHitCollection const&, Collections);
+    void runOnGpuUncalibRecHits(EcalUncalibratedRecHitCollection const&, Collections);
+    void runOnCpuRecHits(EcalRecHitCollection const&, Collections);
+    void runOnGpuRecHits(EcalRecHitCollection const&, Collections);
+
+  private:
+    void setParams(edm::ParameterSet const&) override;
+
+    bool runGpuTask_;
+    bool gpuOnlyPlots_;
+    std::vector<int> uncalibOOTAmps_;
+
+    EBDigiCollection const* EBCpuDigis_;
+    EEDigiCollection const* EECpuDigis_;
+
+    EcalUncalibratedRecHitCollection const* EBCpuUncalibRecHits_;
+    EcalUncalibratedRecHitCollection const* EECpuUncalibRecHits_;
+
+    EcalRecHitCollection const* EBCpuRecHits_;
+    EcalRecHitCollection const* EECpuRecHits_;
+  };
+
+  inline bool GpuTask::analyze(void const* p, Collections collection) {
```

Comment:
CMS code rule 2.12 says to not use single character variable names except for loop indices. Please rename `p`.